### PR TITLE
Fix Lua IDE breakpoint stall and failure to register self as host

### PR DESCRIPTION
**Pull request by AMZN-puvvadar**
Addresses https://github.com/o3de/o3de/issues/9075

The root cause of the issue is the manner in which Lua IDE halts the Editor when a breakpoint is hit. It does this by entering a while loop that only looks for other debug messages and thread yields. This unfortunately causes all network traffic to cease since network updates are sourced off the main thread's SystemTickBus.

There are a few potential fixes here
1. Allow SystemTickBus to operate in this debug state. Unfortunately, SystemTickBus also triggers things like high priority entity spawning so this did not seem appropriate for a stopped debug state.
2. Create a manual ForceUpdate function to allow AzNetworking to process traffic without SystemTickBus.
3. Move NetworkSystemComponent's updates off the main thread and SystemTickBus entirely.

I implemented the second which works thanks to the while loop having the necessary machinery to to pull TmMessages manually (DispatchMessages' purpose in the loop).

There is a good chance this code will be refactored entirely in the near future.

This change ALSO addresses an issue in which the Lua IDE would startup as a loopback due to TargetManagementComponent being on the Application Entity vs a Module Entity. StandaloneToolsApplication now accounts for both possibilities.

Signed-off-by: puvvadar <puvvadar@amazon.com>

### CODE REVIEW

**Comment by AMZN-puvvadar** [comment]
Per a comment in https://github.com/o3de/o3de/issues/9075, Step debug functions where stalling out in the Lua IDE with a loading bar. This is caused by under the hood breakpoints generated for the Step functions looking for .lua assets instead of .luac. I'm not familiar with how that all works under the hood so I added a quick piece of fix up to deal with it in breakpoint handling.

**Review by kberg-amzn — APPROVED** [review]

File: Code/Tools/LuaIDE/Source/LUA/LUAEditorContext.cpp [hunk 1]
```diff
@@ -1800,6 +1800,13 @@ namespace LUAEditor
         AZStd::string absolutePath = relativePath.substr(1);
         EBUS_EVENT(AzToolsFramework::AssetSystemRequestBus, GetFullSourcePathFromRelativeProductPath, absolutePath, absolutePath);
 
+        // If finding a .lua fails, attempt the equivalent .luac
```
[the remark this answers is not in the document]
Comment (AMZN-puvvadar) [reply Code/Tools/LuaIDE/Source/LUA/LUAEditorContext.cpp:1803]: It chops off the first character, which seems to be an '@' generally and then passes it to GetFullSourcePathFromRelativeProductPath as both the param and out value. The more accurate read would probably be

```
AZStd::string absolutePath;
AZStd::string formattedRelativePath = relativePath.substr(1);
EBUS_EVENT(AzToolsFramework::AssetSystemRequestBus, GetFullSourcePathFromRelativeProductPath, formattedRelativePath , absolutePath);
```

**Review by lmbr-pip — APPROVED** [review]
Thanks for fixing this, not sure what else we can do in the short term. 

Could we make a feature request for the correct way of fixing this, so we don't forget the learnings?

**Comment by galibzon** [comment]
I just validated this PR by cherry-picking on top of latest `development` branch and lua debugging seems to be working again.

**Review by carlitosan — APPROVED** [review]

**Review by lsemp3d — APPROVED** [review]

**Pull request merged**